include/appfwk/DAQModule.hpp

```diff
@@ -99,10 +156,34 @@ class DAQModule : public NamedObject
    *  Non-accepted commands or failure should return an ERS exception
    * indicating this result.
    */
-  virtual void execute_command(const std::string& cmd, const std::vector<std::string>& args) = 0;
+  void execute_command(const std::string& name, const std::vector<std::string>& args = {});
+
+  std::vector<std::string> get_commands() const;
+
+  bool has_command(const std::string name) const;
```

Comment:
for the same reason this also should go to virtual

Reply:
maybe let's add a reference to be consistent everywhere

Reply:
Same answer as above. This is meant not to be virtual.

Reply:
Why? I'm not saying it has to be encouraged, but still not providing freedom for flexibility is a dangerous thing in a framework. The usage case for an overload will be that depending on the state of DAQ module the same command  can go to different functions.

Reply:
In all practical scenarios, because the additional freedom adds little in terms of flexibility while increasing the risk for users to bypass command creation. 
Allowing the same command having a different behaviour depending on the state, why wouldn't one handle such case in the command method?

Reply:
simple code modularity. You write the simplest function you can and if it's necessary you combine them as necessary.

Reply:
Let's get a couple of things straight about the requirements on the framework, which still seem unclear.
The goal of the framework is to provide as much flexibility as possible, within the robustness and predictability requirements.
When it comes to module commands (stateless actions, let's not mix things up with FSM transitions), this translates into the module 
* having the list of commands defined by the end of initialisation
* the behaviour of each command predictable at all times
* the list of commands accessible by the caller (`DAQProcess` in our case)
* a user API that **enforces** within reason the previous requirements to be met.

A command registry is a typical pattern to achieve this (see `XDAQ` and others). 
Code modularity shall be encourage as much as possible within these boundaries.
So,
* mutable command behaviour is actually **not** desirable
* mutable behaviour of `execute_command` is not desirable
* state-aware commands are an option, they are not in scope for v1 and they will be equally regulated.
* there is still a lot of rope for users to hang themselves, if they have a deep desire to, for instance with hand-crafted ifs. If they do, they will be shown the wrongness of their ways.